### ISSUE TITLE
Rule MD077 affected by level of indentation

### ISSUE THREAD

**Issue by eread** [rvben/rumdl#739]
### rumdl version

0.2.34

### Installation method

mise

### Bug description

Rule MD077 (https://github.com/rvben/rumdl/blob/c0f492bd612f1d3031ea46d40f6bf827f7fed115/docs/md077.md) checks for incorrect list continuation indentation.

However, it appears that the rule is affected by the level of indentation at which the list continuation occurs.

### Steps to reproduce

1. Create a Markdown file called `test.md` with this content:

   ```markdown
   ---
   title: Heading
   ---
   
   Test Markdown.
   
   1. Level 1 item.
   1. Level 1 item.
   1. Level 1 item:
      - Level 2 item.
      - Level 2 item.
   
      Level 1 list continuation.
   1. Level 1 item.
   
   Test Markdown.
   
   1. Level 1 item.
   1. Level 1 item:
      1. Level 2 item.
      1. Level 2 item.
      1. Level 2 item:
         - Level 3 item.
         - Level 3 item.
   
         Level 2 list continuation.
   1. Level 1 item.
   ```
1. Run `rumdl check test.md` with no additional configuration.

### Expected behavior

I'd expect no violation of MD077 from the Markdown snippet.

### Actual behavior

rumdl returns:

```shell
test.md:26:1: [MD077] Continuation line over-indented (expected 3, found 6) [*]
```

It appears the rule doesn't realise which list it's continuing from. It might be the level 1 list (rather than the level 2 list I have), but I'd expect either "continuing the level 1 list" or "continuing the level 2 list" to both be valid options.

### Configuration

No custom configuration.

### Operating system

macOS

**Comment by rvben** [comment]
Fixed in v0.2.42 - MD077 no longer mis-attributes a nested list continuation to the wrong list level. Your repro now reports no violations. Thanks @eread.
